src/core/unit.c

```diff
@@ -4635,7 +4636,7 @@ int unit_write_setting(Unit *u, UnitWriteFlags flags, const char *name, const ch
                 return r;
         q = NULL;
 
-        strv_uniq(u->dropin_paths);
+        strv_sort_uniq(u->dropin_paths);
```

Comment:
This seems wrong. We need to sort this based on the actual loading priorities (/etc/, /run/, /usr/local/, /usr/), not lexical order.